src/components/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,151 @@
+  onSelect: (event: unknown) => void;
+  onSelectCapture: (event: unknown) => void;
+  hint?: string;
+}
+
+const Textarea: L3Component<TextareaProps, unknown> = forwardRef(
+  (
+    {
+      placeholder = "",
+      autoComplete = "off",
+      initialValue,
+      value,
+      onChange,
+      onBlur = NOOP,
+      onFocus = NOOP,
+      onKeyDown = NOOP,
+      onChangeCapture,
+      debounceRate = 0,
+      autoFocus = false,
+      disabled = false,
+      readonly = false,
+      setRef = NOOP,
+      trim = false,
+      id = "textarea",
+      required = false,
+      name,
+      cols = 40,
+      rows = 7,
+      maxLenght = 1200,
+      minLenght,
+      onInvalid,
+      onInvalidCapture,
+      onSelect,
+      onSelectCapture,
+      hint,
+      validation = null
+    },
+    ref
+  ) => {
+    const inputRef = useRef(null);
+    const { inputValue, onEventChanged, clearValue } = useDebounceEvent({
+      delay: debounceRate,
+      onChange,
+      initialStateValue: value,
+      trim
+    });
+
+    const mergedRef = useMergeRefs({ refs: [ref, inputRef, setRef] });
+    useEffect(() => {
+      if (inputRef.current && autoFocus) {
+        const animationFrame = requestAnimationFrame(() => inputRef.current.focus());
+        return () => cancelAnimationFrame(animationFrame);
+      }
+    }, [inputRef, autoFocus]);
+
+    return (
+      <div
+        className={cx(styles.mainWrapper, {
+          [styles.status_error]: validation && validation.status === "error",
+          [styles.status_success]: validation && validation.status === "success",
+          [styles.disabled]: disabled === true
+        })}
+      >
+        {disabled && <div style={{ position: "absolute" }}>disabled</div>}
+        <label htmlFor={`${id}`} className={styles.labelTop}>
+          {inputValue ? inputValue.length : 0}/{maxLenght}
+        </label>
+        {inputValue && (
+          <div className={styles.clearIcon}>
+            <IconButton
+              size={"xxs"}
+              ariaLabel="Remove"
+              hideTooltip
+              icon={CloseSmall}
+              onClick={clearValue}
+              kind={IconButton.kinds.SECONDARY}
+            />
+          </div>
+        )}
+        <textarea
+          ref={mergedRef}
+          id={`${id}`}
+          className={styles.textarea_input}
+          value={inputValue}
+          placeholder={placeholder}
+          autoComplete={autoComplete}
+          autoFocus={autoFocus}
+          defaultValue={initialValue}
+          cols={cols}
+          rows={rows}
+          disabled={disabled}
+          maxLength={maxLenght}
+          minLength={minLenght}
+          name={name}
+          onBlur={onBlur}
+          onFocus={onFocus}
+          onKeyDown={onKeyDown}
+          onChange={onEventChanged}
+          onChangeCapture={onChangeCapture}
+          onInvalid={onInvalid}
+          onInvalidCapture={onInvalidCapture}
+          onSelect={onSelect}
```

Comment:
please check the props some of these props is not relevant to the textarea,  for example, select is not the correct prop for it,
this more details here 
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea